### ISSUE TITLE
Choking on version strings from setuptools_scm

### ISSUE THREAD

**Issue by jonathanunderwood** [briancurtin/deprecation#9]
When a project uses the `setuptools_scm` package to create version number strings fromSCM automatically, the version strings look like: `0.11.2.dev208+g1e95728.d20171223`. Using these version strings with the deprecation module causes things to fail:

```python
tests/block/test_block.py:1: in <module>
    import lz4.block
.tox/py/lib/python3.6/site-packages/lz4/__init__.py:14: in <module>
    details="Use the lz4.library_version_number function instead")
.tox/py/lib/python3.6/site-packages/deprecation.py:131: in deprecated
    current_version = version.StrictVersion(current_version)
/usr/lib64/python3.6/distutils/version.py:40: in __init__
    self.parse(vstring)
/usr/lib64/python3.6/distutils/version.py:137: in parse
    raise ValueError("invalid version number '%s'" % vstring)
E   ValueError: invalid version number '0.11.2.dev208+g1e95728.d20171223'
```
This is caused by this piece of code:
```python
@deprecated(deprecated_in="0.14", removed_in="1.0",
            current_version=VERSION,
            details="Use the lz4.library_version_number function instead")
```

Removing the `current_version` argument resolves the problem but of course then you lose the ability to raise exceptions when a function hasn't been removed.

It would probably be better to use  [packaging.version.parse](https://packaging.pypa.io/en/latest/version/#packaging.version.parse) for version string comparisons, which is the standard used by setuptools, I think.

**Comment by jonathanunderwood** [comment]
Oh, wow, I just noticed that PR #8 actually implements what's needed to fix this.

**Comment by briancurtin** [comment]
I think #8 looks good, though I want to look a little further into it just so I know what it's doing. I'm currently on vacation until January 8, but may have time next week to look at this.

**Comment by briancurtin** [comment]
Released as 1.1: https://pypi.python.org/pypi/deprecation/1.1

Thanks for the report and thanks @debosmit for the PR!
